### PR TITLE
Strip symbols from production binary (rust >= 1.59)

### PR DESCRIPTION
**Pull request by grtlr**
# Description of change

This should produce even smaller binaries, but requires Rust >= 1.59.

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

**Review by thibault-martinez — APPROVED** [review]

**Comment by grtlr** [comment]
@thibault-martinez Here is a test of what happens when `strip = symbols`, because there were some concerns.

TLDR: Line numbers are still in there, only the backtrace contains less and less information.

### Example code
``` rust
fn main() {
    let mut hello = Some("Hello").to_owned();
    println!("{}", hello.unwrap());
    hello = None;
    println!("{}", hello.unwrap());
}
```
---
`RUST_BACKTRACE=1 cargo run`:
``` sh
Hello
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/main.rs:5:26
stack backtrace:
	 0: rust_begin_unwind
	           at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
	 1: core::panicking::panic_fmt
	           at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
	 2: core::panicking::panic
	           at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:48:5
	 3: core::option::Option<T>::unwrap
	           at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/option.rs:729:21
	 4: strip_test::main
	           at ./src/main.rs:5:20
	 5: core::ops::function::FnOnce::call_once
	           at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/ops/function.rs:227:5
	note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
---
`RUST_BACKTRACE=1 cargo run --release`:
``` sh
Hello
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/main.rs:5:26
stack backtrace:
	 0: rust_begin_unwind
	           at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/std/src/panicking.rs:498:5
	 1: core::panicking::panic_fmt
	           at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:116:14
	 2: core::panicking::panic
	           at /rustc/9d1b2106e23b1abd32fce1f17267604a5102f57a/library/core/src/panicking.rs:48:5
	 3: strip_test::main
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```
---
`RUST_BACKTRACE=1 cargo run --profile production` (`strip = symbols`):
``` sh
Hello
thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', src/main.rs:5:26
stack backtrace:
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by Dr-Electron** [comment]
Thanks for the insights.
The binary size would also be interesting imo :)

**Review by Adam-Gleave — APPROVED** [review]
